generator/main.ts

```diff
@@ -72,6 +74,52 @@ async function generateInterface(
   });
 }
 
+async function generateEnumTypes(
+    env: nunjucks.Environment, output_path: string,
+    definitions: IDLDefinition[]) {
+  const [enum_type_header_tmpl, enum_type_cpp_tmpl] = await Promise.all([
+    file.read(path.resolve(TEMPLATE_DIR, 'enum_types_header.njk')),
+    file.read(path.resolve(TEMPLATE_DIR, 'enum_types_cpp.njk')),
+  ]);
+
+  const enum_type_header_path = path.resolve(output_path, 'enum_types.h');
+  const enum_type_cpp_path = path.resolve(output_path, 'enum_types.cpp');
+
+  let idl_enum: IDLDefinition[] = [];
+  definitions.forEach(async (definition) => {
+    if (definition.isIDLEnum()) {
+      idl_enum.push(definition);
+    }
+  });
+
+  await file.write(
+      enum_type_header_path,
+      env.renderString(enum_type_header_tmpl, {enums: idl_enum}));
+  await file.write(
+      enum_type_cpp_path,
+      env.renderString(enum_type_cpp_tmpl, {enums: idl_enum}));
+}
+
+// TODO(hwansueng) :: this function should be improved.
```

Comment:
nit: // TODO(hwanseung): This function should be improved.

Reply:
done